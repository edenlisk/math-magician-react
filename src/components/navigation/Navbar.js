import { Link } from 'react-router-dom';
import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';

const Navbar = () => (
  <nav className="container-fluid d-flex justify-content-end">
    <div className="container-fluid">
      <h4>Math Magician!</h4>
    </div>
    <Link className="mx-1 fw-bold" to="/">Home</Link>
    <span>|</span>
    <Link className="mx-1 fw-bold" to="/calculator">Calculator</Link>
    <span>|</span>
    <Link className="mx-1 fw-bold" to="/quote">Quote</Link>
  </nav>
);

export default Navbar;
