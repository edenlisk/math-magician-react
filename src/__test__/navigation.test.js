import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import Navbar from '../components/navigation/Navbar';
import '@testing-library/jest-dom';

describe('Test navigation bar', () => {
  test('Test Navbar', () => {
    render(<Navbar />);
    expect(screen.getByTestId('header')).toHaveTextContent('Math Magician!');
  });

  test('It should render Navbar correctly', () => {
    const tree = TestRenderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
