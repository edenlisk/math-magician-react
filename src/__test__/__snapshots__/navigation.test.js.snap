// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test navigation bar It should render Navbar correctly 1`] = `
<nav
  className="container-fluid d-flex justify-content-end"
>
  <div
    className="container-fluid"
  >
    <h4
      data-testid="header"
    >
      Math Magician!
    </h4>
  </div>
  <a
    className="mx-1 fw-bold"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <span>
    |
  </span>
  <a
    className="mx-1 fw-bold"
    href="/calculator"
    onClick={[Function]}
  >
    Calculator
  </a>
  <span>
    |
  </span>
  <a
    className="mx-1 fw-bold"
    href="/quote"
    onClick={[Function]}
  >
    Quote
  </a>
</nav>
`;
